src/base/tree.rs

```diff
@@ -116,14 +135,27 @@ impl<'dt> DevTree<'dt> {
         if Self::read_totalsize(buf)? < buf.len() {
             Err(DevTreeError::ParseError)
         } else {
-            let ret = Self { buf };
-            // Verify required alignment before returning.
-            verify_offset_aligned::<u32>(ret.off_mem_rsvmap())?;
-            verify_offset_aligned::<u32>(ret.off_dt_struct())?;
-            Ok(ret)
+            Self::from_safe_slice(buf)
         }
     }
 
+    /// Construct the parseable DevTree object from a raw byte pointer
+    ///
+    /// # Safety
+    ///
+    /// Callers of this method the must guarantee the following:
+    ///
+    /// - The passed buffer is 32-bit aligned.
+    /// - The passed buffer is exactly the length returned by [`Self::read_totalsize()`]
```

Comment:
Safety list should be changed to:

```
- The passed pointer is 32-bit aligned.
```

The second point isn't necessary since I assume the user won't be reading the totalsize themselves.

Reply:
My bad, copypasta oversight. Fixed in 08aaffe